pkg/proxy/ipvs/proxier.go

```diff
@@ -966,6 +967,17 @@ func (proxier *Proxier) OnNodeUpdate(oldNode, node *v1.Node) {
 		return
 	}
 
+	// kube-proxy in LocalModeNodeCIDR mode may cache stale Node.PodCIDR.
+	// It should restart on those situations to reconcile the state.
+	// Ref: https://issues.k8s.io/111321
+	if proxier.localDetector.IsImplemented() && proxier.localDetector.Mode() == proxyconfigapi.LocalModeNodeCIDR {
+		nodeLocalCIDR := proxier.localDetector.IfLocal()[1]
+		if !utilproxy.NodeContainsPodCIDR(node, nodeLocalCIDR) {
+			klog.ErrorS(nil, "Inconsisten statu, kube-proxy configed Local CIDR not present on Node.Spec.PodCIDRs", "node", klog.KObj(node), "local CIDR", nodeLocalCIDR, "PodCIDRs", node.Spec.PodCIDRs)
```

Comment:
Match this log message with the one above.

Also, this code looks to be the same, we might want to factor it out to a func? (name it MaybeRestartOnInconsistentPodCIDR()?)

Reply:
I thought about that, the reasons I didn't do it is because these are 7 lines of code with a lot of deps on `proxier` methods ... the refactor to a func is not going to save more lines and will make it harder to read IMHO ...

Reply:
to be honest, I like your idea, but with current state of the code on both proxies, I don't know if it is better to keep them isolated or to try to consolidate them